### PR TITLE
Task 4 on Bash realization

### LINKED ISSUE #7: Task4_Bash
Create a script that takes file path and text as arguments, but these arguments are defined by keys (-p path -s text), 
and returns list of all files that contain the specified text in the given directory.

### PR DESCRIPTION
**Pull request by Legamanchik**
Task 4 on bash

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

### CODE REVIEW

**Review by AndriyGalias — COMMENTED** [review]
Not critical here, but you do violation of open-closed principle using cases

**Review by AndriyGalias — APPROVED** [review]

File: Task_4.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+while getopts "p:t:" flag; do
+    case "${flag}" in
+        p) path="${OPTARG}" ;;
+        t) text="${OPTARG}" ;;
+    esac
+done
+
+if [ $# -eq 0 ]; then
```
Comment (arkma) [inline Task_4.sh]:
```suggestion
if [ $# -neq 2 ]; then
```
You should check if two parameters were provided. Alternatively, create a function for script usage and check them later.

Comment (arkma) [reply Task_4.sh]: In addition, try to avoid nested if else structures for better readability.

Comment (arkma) [reply Task_4.sh]: Also change script name to 'task4.sh'.

File: Task_4.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        exit 1
+    elif [ ! -e "$path" ]; then
+        echo "Error: $path path does not exist"
+        exit 1
+    elif [ -z "$text" ]; then
+        echo "Error: -t text is empty"
+        exit 1
+        
+    else
+        grep -R "$text" "$path"
```
Comment (arkma) [inline Task_4.sh]: It seems to me that you forgot to use the '-l' parameter for grep. In your case, you print all lines where `$text` matches. Try this one: `grep -R -l "$text" "$path"` to print only the file name.

**Review by arkma — CHANGES_REQUESTED** [review]

**Review by arkma — APPROVED** [review]

**Review by OlehFedyshyn — APPROVED** [review]
All good

**Pull request merged**